src/Enchantments.cpp

```diff
@@ -242,6 +242,323 @@ bool cEnchantments::operator !=(const cEnchantments & a_Other) const
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsShovel(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchEfficiency) &&
+				(a_EnchantmentID != cEnchantments::enchSilkTouch) &&
+				(a_EnchantmentID != cEnchantments::enchFortune) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsPickaxe(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchEfficiency) &&
+				(a_EnchantmentID != cEnchantments::enchSilkTouch) &&
+				(a_EnchantmentID != cEnchantments::enchFortune) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsHoe(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+	}
+
+	else if (ItemCategory::IsArmor(a_ItemType))
+	{
+		if (ItemCategory::IsHelmet(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchRespiration) &&
+				(a_EnchantmentID != cEnchantments::enchAquaAffinity) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsChestPlate(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsLeggings(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsBoots(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFeatherFalling) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchDepthStrider) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+	}
```

Comment:
basically what I see is you have a list of items and then for each item the list of enchantments (valid or invalid, depending on how you base the logic).

If I am on the right path with the above, I can start to think about how to build it?